client/post-editor/editor-location/map.jsx

```diff
@@ -0,0 +1,109 @@
+/** @format */
+/* global google */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import qs from 'querystring';
+
+export default class extends React.Component {
+	static displayName = 'EditorLocationMap';
+
+	static propTypes = {
+		coordinates: PropTypes.array,
+		onError: PropTypes.func,
+		onSelect: PropTypes.func,
+	};
+
+	static defaultProps = {
+		coordinates: [],
+		onError: () => {},
+		onSelect: () => {},
+	};
+
+	componentDidMount() {
+		// Connect the initMap() function within this class to the global window
+		// context, so Google Maps can invoke it.
+		window.initMap = this.initMap.bind( this );
+
+		const src =
+			'https://maps.googleapis.com/maps/api/js?' +
+			qs.stringify( {
+				key: 'INSERT-API-KEY-HERE',
+				callback: 'initMap',
+			} );
+
+		// Asynchronously load the Google Maps script, passing in the callback reference.
+		// This dumps the `google` var into the global scope, which is not fun, but the only way.
+		loadJS( src );
+	}
+
+	initMap() {
+		this.map = new google.maps.Map( this.mapRef, {
+			zoom: 12,
+		} );
+
+		// This event listener will call props.onSelect(), which will cause a
+		// component re-render, which will cause componentDidUpdate to run,
+		// setting the map marker on the clicked location.
+		this.map.addListener( 'click', event => {
+			this.props.onSelect( {
+				geometry: {
+					location: {
+						lat: event.latLng.lat(),
+						lng: event.latLng.lng(),
+					},
+				},
+			} );
+		} );
+	}
+
+	componentDidUpdate() {
+		if ( this.map && this.props.coordinates ) {
```

Comment:
could `this.props.coordinates` ever be an empty array here? Also should we consider deconstructing `const { coordinates } = this.props;` to clean things up a a bit?

Reply:
Good point. The `EditorLocation` component won't render the map if coordinates is falsy so a falsy check here is pointless. So far I've only seen `this.map` undefined, but it'd best to check the length of the coordinates.